packages/yew-macro/src/html_tree/html_component.rs

```diff
@@ -25,13 +25,16 @@ impl Parse for HtmlComponent {
         };
 
         if trying_to_close() {
-            return match input.parse::<HtmlComponentClose>() {
-                Ok(close) => Err(syn::Error::new_spanned(
-                    close.to_spanned(),
-                    "this closing tag has no corresponding opening tag",
-                )),
-                Err(err) => Err(err),
-            };
+            let close = input.parse::<HtmlComponentClose>();
+            if !is_ide_completion() {
+                return match close {
```

Comment:
Would it make sense to move `is_ide_completion` call to root of the macro where `syn::Error` is converted to `compile_err!`? That way all the cases of compilation failure will covered, while avoiding adding the if conditional at every error return

Reply:
Hmm, but what can we do with `syn::Error` except converting it to `compile_err!`? Here we skipping returning with an error an continuing the parsing. If we return with an error, the parsing is abandoned, so we can do nothing except returning `compile_err!`, I guess

Reply:
The macro needs to expand to actual results (even if incomplete)? I was thinking maybe we could return an empty TokenStream and such so the output doesn't return a compile error. `Html::default()` is valid return type for every macro. Perhaps that can be returned?